File: backend/farm-service/src/main/java/ase_pr_inso_01/farm_service/models/Field.java
package ase_pr_inso_01.farm_service.models;

import ase_pr_inso_01.farm_service.models.enums.SeedType;
import lombok.Getter;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Date;


@Document(collection = "fields")
@Getter
@Setter
public class Field {
    @Id
    private Integer id;
    private String status; //TODO: Enum
    private String seedType;
    private Date plantedDate;
    private Date harvestDate;
    private String growthStage; //TODO: Enum
}

File: frontend/src/app/dtos/field.ts
import { FieldStatus } from "../models/FieldStatus";
import { GrowthStage } from "../models/GrowthStage";
import { SeedType } from "../models/Seed";

export class FieldDetailsDto {
  id!: number;
  status!: string; //TODO: enum / type
  seedType?: string; //TODO: enum / type
  plantedDate?: Date;
  harvestDate?: Date;
  growthStage?: GrowthStage;
}

export class FieldUpdateDto {
  id!: number;
  status!: string; //TODO: enum / type
  seedType!: string; //TODO: enum / type
  plantedDate!: Date;
  growthStage?: GrowthStage;
}

export class FieldCreateDto {
  id!: number;
  status: FieldStatus = 0; // Default to Empty
}
